cognite/client/_api/raw.py

```diff
@@ -339,29 +342,89 @@ def __call__(
         Args:
             db_name (str): Name of the database
             table_name (str): Name of the table to iterate over rows for
-            chunk_size (int | None): Number of rows to return in each chunk. Defaults to yielding one row a time.
+            chunk_size (int | None): Number of rows to return in each chunk. Defaults to yielding one row a time. Note: When used together with 'partitions'
+                the default is 10000 (matching the API limit) and there's an implicit minimum of 1000 rows.
             limit (int | None): Maximum number of rows to return. Defaults to return all items.
             min_last_updated_time (int | None): Rows must have been last updated after this time (exclusive). ms since epoch.
             max_last_updated_time (int | None): Rows must have been last updated before this time (inclusive). ms since epoch.
             columns (list[str] | None): List of column keys. Set to `None` for retrieving all, use [] to retrieve only row keys.
+            partitions (int | None): Retrieve rows in parallel using this number of workers. Defaults to 1 (i.e. _no concurrency_).
+                The setting is capped at ``global_config.max_workers`` and can not be used with a finite limit. To prevent unexpected
+                problems and maximize read throughput, check out `concurrency limits in the API documentation. <https://developer.cognite.com/api#tag/Raw/#section/Request-and-concurrency-limits>`_
 
         Returns:
-            Iterator[Row] | Iterator[RowList]: No description.
+            Iterator[Row] | Iterator[RowList]: An iterator yielding the requested rows (or row) in chunks.
         """
-        return self._list_generator(
-            list_cls=RowList,
-            resource_cls=Row,
-            resource_path=interpolate_and_url_encode(self._RESOURCE_PATH, db_name, table_name),
+        if partitions is None or not is_unlimited(limit):
+            return self._list_generator(
+                list_cls=RowList,
+                resource_cls=Row,
+                resource_path=interpolate_and_url_encode(self._RESOURCE_PATH, db_name, table_name),
+                chunk_size=chunk_size,
+                method="GET",
+                limit=limit,
+                filter={
+                    "minLastUpdatedTime": min_last_updated_time,
+                    "maxLastUpdatedTime": max_last_updated_time,
+                    "columns": self._make_columns_param(columns),
+                },
+            )
+        return self._list_generator_concurrent(
```

Comment:
This execution path is not covered by any test, including `_list_generator_concurrent`. Shouldn't there be a simple test for it?

Reply:
Yes :hide: